mlflow/optuna/storage.py

```diff
@@ -271,12 +273,45 @@ def get_study_id_from_name(self, study_name: str) -> int:
         # Flush all batches to ensure we have the latest data
         self.flush_all_batches()
 
-        runs = self._search_run_by_name(study_name)
+        runs = self._search_runs_by_name(study_name)
         if len(runs):
             return runs[0].info.run_id
         else:
             raise Exception(f"Study {study_name} not found")
 
+    def study_exists(self, study_name: str) -> bool:
+        """Check if a study with the given name exists.
+
+        Args:
+            study_name: The name of the study to check
+
+        Returns:
+            True if the study exists, False otherwise
+        """
+        # Flush all batches to ensure we have the latest data
+        self.flush_all_batches()
+
+        runs = self._search_runs_by_name(study_name)
```

Comment:
```suggestion
        runs = self._search_runs_by_name(study_name, max_results=1)
```

can we add `max_results` to avoid unnecessary conversion?